Wrong class used in turbo streams

htmlshaman1 opened the pull request:
TurboBundle::STREAM_FORMAT doesn't exist, that static variable was moved to TurboStreamResponse::STREAM_FORMAT.
Also the stream didn't work for me unless I changed:
$this->render('task/success.stream.html.twig', ['task' => $task]); to $this->render('task/success.stream.html.twig', ['task' => $task], new TurboStreamResponse());

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no <!-- please update src/**/CHANGELOG.md files -->
| Tickets       | Fix #... <!-- prefix each issue number with "Fix #", no need to create an issue if none exist, explain below instead -->
| License       | MIT

<!--
Replace this notice by a short README for your feature/bugfix. This will help people
understand your PR and can be used as a start for the documentation.

Additionally (see https://symfony.com/releases):
 - Always add tests and ensure they pass.
 - Never break backward compatibility (see https://symfony.com/bc).
 - Features and deprecations must be submitted against branch main.
-->

htmlshaman1 committed:
Wrong class used in turbo streams

TurboBundle::STREAM_FORMAT doesn't exist, that static variable was moved to TurboStreamResponse::STREAM_FORMAT.
Also the stream didn't work for me unless I changed:
$this->render('task/success.stream.html.twig', ['task' => $task]); to $this->render('task/success.stream.html.twig', ['task' => $task], new TurboStreamResponse());

weaverryan commented:
Hi!

Sorry about this. The documentation IS correct but not for the currently-released version. Let me be more clear:

* v2.0  (current version) uses `TurboStreamResponse` (as you've seen)
* v2.1 (unreleased) uses what you see in the documentation

What we need in the docs (and should have added when this change was originally made) was a `versionadded:: 2.1` tag that explained the change and how it worked before. For an example, see https://github.com/symfony/ux/blob/2.x/src/LiveComponent/src/Resources/doc/index.rst#L59

Would you be able to revert your changes by add a nice `versionadded` note explaining things?

Thanks!

htmlshaman1 committed:
Added a message about it was done in version 2.0

I added a helpful message trying to explain that the old version used a different class, and returned a stream response.

htmlshaman1 reviewed (commented):
Added a note about the class change from version 2.0 to 2.1

htmlshaman1 commented:
I added a brief message about the previous version using a different class, and returning a streaming response. If you can't use it, that's alright. I never knew contributing to the docs was so easy and smooth.

weaverryan commented on src/Turbo/Resources/doc/index.rst:
```diff
@@ -326,18 +326,23 @@ Let's discover how to use Turbo Streams to enhance your `Symfony forms`_::
-                    $request->setFormat(TurboStreamResponse::STREAM_FORMAT);
-                    return $this->render('task/success.stream.html.twig', ['task' => $task], new TurboStreamResponse());
+                    $request->setFormat(TurboBundle::STREAM_FORMAT);
+                    return $this->render('task/success.stream.html.twig', ['task' => $task]);
```
Could you also revert these changes? We will only show the NEW way

weaverryan commented on src/Turbo/Resources/doc/index.rst:
```diff
@@ -326,18 +326,23 @@ Let's discover how to use Turbo Streams to enhance your `Symfony forms`_::
-            // Symfony 5.3+
+  .. versionadded:: 2.1
+  
+  In versions prior to 2.1, TurboStreamResponse::STREAM_FORMAT was used instead of TurboBundle::STREAM_FORMAT. Also, one had to return a new TurboStreamResponse() object as the third argument for this->render().
```
Let's move this above the code block, so people definitely see it.

Also, add manual line breaks around the 72nd character.

And put "ticks" (these characters: \`) around the technical words like `TurboStreamResponse::STREAM_FORMAT`.

weaverryan reviewed (commented):
A few tweaks - thanks for making the changes :)

htmlshaman1 committed:
I think I got it

Sorry, Ryan this is my first time trying to contribute to the docs. I couldn't figure out how to revert a change on GitHub's interface. I think I got it right.
Anyway, this has been kind of fun. Thanks! I'm going to read https://symfony.com/doc/current/contributing/documentation/overview.html as well.

htmlshaman1 reviewed (commented):
I think I have it.

htmlshaman1 reviewed (commented):
Moved helpful message above code

weaverryan committed:
minor cleanups

weaverryan commented:
Thanks @htmlshaman1!
